### PR TITLE
FISH-477 Deployment group properties are now resolved

### PR DESCRIPTION
**Pull request by Cousjava**
## Description
This is a bug fix

### Testing Performed
Manual tests, see Jira for reproducer

### Testing Environment
Zulu JDK 1.8_212 on Ubuntu 20.04 with Maven 3.6.3

### CODE REVIEW

**Comment by Cousjava** [comment]
Jenkins test please

File: nucleus/core/kernel/src/main/java/com/sun/enterprise/v3/server/SystemTasksImpl.java [hunk 1]
```diff
@@ -173,15 +174,20 @@ private void setSystemPropertiesFromDomainXml() {
...
+        // 4. domain
         // so we need to add System Properties in *reverse order* to get the
         // right precedence.
 
         List<SystemProperty> domainSPList = domain.getSystemProperty();
         List<SystemProperty> configSPList = getConfigSystemProperties();
         Cluster cluster = server.getCluster();
         List<SystemProperty> clusterSPList = null;
-
+        List<DeploymentGroup> depGroups = server.getDeploymentGroup();
```
Comment (Pandrex247) [inline nucleus/core/kernel/src/main/java/com/sun/enterprise/v3/server/SystemTasksImpl.java:186]: Can this be null?
Having trouble reproducing atm

Comment (Cousjava) [reply nucleus/core/kernel/src/main/java/com/sun/enterprise/v3/server/SystemTasksImpl.java:186]: This cannot be null, but the list may be of length 0.

Comment (Cousjava) [reply nucleus/core/kernel/src/main/java/com/sun/enterprise/v3/server/SystemTasksImpl.java:186]: This cannot be null, but is can be a 0-sized list.

**Comment by jGauravGupta** [comment]
Jenkins test please

**Review by Pandrex247 — CHANGES_REQUESTED** [review]
This gets validated twice, once against the DAS config, and once against the target config - meaning you need to have the property set on both configs for it to pick it up. Ideally this should validate it against just the target config.

**Review by Pandrex247 — APPROVED** [review]

**Comment by Cousjava** [comment]
Jenkins test please

**Pull request merged**